clang/lib/Driver/Driver.cpp

```diff
@@ -608,6 +608,20 @@ Driver::OpenMPRuntimeKind Driver::getOpenMPRuntime(const ArgList &Args) const {
   return RT;
 }
 
+static bool isValidSYCLTriple(llvm::Triple T) {
+  // Check for invalid SYCL device triple values.
+  // Non-SPIR arch.
+  if (T.getArch() == llvm::Triple::UnknownArch || !T.isSPIR())
```

Comment:
```suggestion
  if (!T.isSPIR())
```
https://github.com/intel/llvm/blob/sycl/llvm/include/llvm/ADT/Triple.h#L697 - current implementation of isSPIR seems to cover first condition as well. Right?

Reply:
You're right.  I have updated.

Reply:
Thanks!